tasks/message/message.go

```diff
@@ -0,0 +1,246 @@
+package message
+
+import (
+	"context"
+	"time"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	"go.opentelemetry.io/otel/api/global"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/lens"
+	messagemodel "github.com/filecoin-project/sentinel-visor/model/messages"
+	"github.com/filecoin-project/sentinel-visor/model/visor"
+	"github.com/filecoin-project/sentinel-visor/storage"
+	"github.com/filecoin-project/sentinel-visor/wait"
+)
+
+const idleSleepInterval = 60 * time.Second   // time to wait if the processor runs out of blocks to process
+const batchInterval = 100 * time.Millisecond // time to wait between batches
+
+var log = logging.Logger("message")
+
+var timeNow = time.Now
```

Comment:
Why?

Reply:
To assist testing so we can use a known time in comparisons. Yes, I know there are no tests in that package... yet :smile:

Reply:
I recommend looking at https://github.com/raulk/clock which is what lotus uses for this

Reply:
Will do although we don't need anything sophisticated yet. Mainly just want to be able to control the time written to completed_at columns.

Reply:
clock provides what you are looking for and I would prefer if we don't duplicate pkgs